app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -189,15 +194,75 @@ private void loadInstanceState(Bundle savedInstanceState) {
     }
 
     /**
-     * Set state that signifies activity was launch from external app.
+     * Set state that signifies activity was launch from external app
      */
     private void processFromExternalLaunch(Bundle savedInstanceState) {
         if (savedInstanceState == null && getIntent().hasExtra(DispatchActivity.WAS_EXTERNAL)) {
             wasExternal = true;
+            processSessionEndpoint();
             sessionNavigator.startNextSessionStep();
         }
     }
 
+    private void processSessionEndpoint() {
+        if (getIntent().hasExtra(SESSION_ENDPOINT_ID)) {
+            Endpoint endpoint = validateIntentForSessionEndpoint(getIntent());
+            if (endpoint != null) {
+                Vector<String> endpointArguments = endpoint.getArguments();
+                Bundle intentArguments = getIntent().getBundleExtra(SESSION_ENDPOINT_ARGUMENTS);
+                EvaluationContext evaluationContext = CommCareApplication.instance().getCurrentSessionWrapper().getEvaluationContext();
```

Comment:
Should we be creating a copy of `evaluationContext` here ?

Reply:
Requests to the session wrapper for an EC should always return a new, clean evaluation context, so I don't think you explicitly need to fork it here unless it will be used in more than one subcontext (IE: If, for example, you needed to evaluate expressions with different sets of argument variables being set).

Reply:
that makes sense. Thanks!